### PR TITLE
[Clouds] Remove old node providers for GCP and Kubernetes

### LINKED ISSUE #3285: Remove deprecated things now that we are >= 0.5
Searching regex `remove.*0.5` across `*.py` shows a few deprecated things we should remove now.

### PR DESCRIPTION
**Pull request by Michaelvll**
We have migrated GCP and Kubernetes to the new provisioner API. It should be good to remote the old node provider codes for them.

Fixes #3285 
- Removed cpu/tpu/gpunode
- Removed env `SKYPILOT_JOB_ID`
- Removed `spot_status`
- Removed  backward compatibility of original `cluster_name` without hash

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch -c test-gcp --cloud gcp --cpus 2 echo hi`
  - [x] `sky autostop -i 0 test-gcp`
  - [x] `sky launch -c test-k8s --cloud kubernetes --cpus 2 echo hi`
  - [x] `sky autostop -i 0 test-k8s --down`
- [x] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/cli.py [hunk 1]
```diff
@@ -5558,8 +5558,6 @@ def local_down():
         click.echo(f'{style.BRIGHT}Local cluster removed.{style.RESET_ALL}')
 
 
-# TODO(skypilot): remove the below in v0.5.
-_add_command_alias_to_group(spot, spot_queue, 'status', hidden=True)
 _deprecate_and_hide_command(group=None,
```
Comment (concretevitamin) [inline sky/cli.py:5561]: The interactive node commands can also be removed?

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks @Michaelvll.

File: sky/cli.py (hunk 1 above)
Comment (Michaelvll) [reply sky/cli.py]: Good point! Removed. Thanks!

**Comment by Michaelvll** [comment]
We just removed the CPU/TPU/GPU node and tested it with `tests/test_smoke.py`. PTAL @concretevitamin  : )

File: sky/cli.py [hunk 2]
```diff
@@ -725,34 +528,8 @@ def _parse_override_params(
...
-
-def _check_resources_match(backend: backends.Backend,
-                           cluster_name: str,
-                           task: 'sky.Task',
-                           node_type: Optional[str] = None) -> None:
+def _check_resources_match(backend: backends.Backend, cluster_name: str,
+                           task: 'sky.Task') -> None:
     """Check matching resources when reusing an existing cluster.
 
     The only exception is when [cpu|tpu|gpu]node -c cluster_name is used with no
```
Comment (concretevitamin) [inline sky/cli.py:535]: Should update?

Comment (concretevitamin) [reply sky/cli.py:535]: Actually, this feels like a shallow func now, can inline.

File: sky/cli.py [hunk 3]
```diff
@@ -846,161 +604,24 @@ def _launch_with_confirm(
...
-    task.set_resources(resources)
-
-    _launch_with_confirm(
-        task,
-        backend,
-        cluster_name,
-        dryrun=False,
-        detach_run=True,
-        no_confirm=no_confirm,
+    # No need to sky.launch again when interactive node is already up.
```
Comment (concretevitamin) [inline sky/cli.py:610]: Remnant?

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks! Should update PR description to include deprecated things :)

File: sky/cli.py (hunk 2 above)
Comment (Michaelvll) [reply sky/cli.py:535]: Good catch! Made it inline

**Pull request merged**